core-processor/src/common.rs

```diff
@@ -250,6 +254,15 @@ pub enum JournalNote {
         /// Decreases gas allowance by that amount, burned for processing try.
         gas_burned: u64,
     },
+    /// Update gas reservations.
+    UpdateGasReservations {
+        /// Message which gas will be reserved from.
+        message_id: MessageId,
+        /// Program ID.
+        program_id: ProgramId,
+        /// Gas reservation map.
+        gas_reserver: GasReserver,
+    },
```

Comment:
Don't you want to split this thing similar to memory and pages notes:
- ReserveGas (this task reserves gas from message only)
- UnreserveGas (this task unreserve gas from message only)
- UpdateGasReservations (this task only updates map inside program structure in storage)

Reply:
What's the point?